search for responses

rosasck opened the pull request:
Ended up hooking up the view, delete buttons, and search all in this PR and stole erics adding self rating column .. sorry peeps :)

rosasck committed:
search for responses

rosasck committed:
merging main in

rosasck committed:
searching

rosasck committed:
Merge branch 'main' of https://github.com/Likhovodov/Simulated-Conversations into Katie/SYMCO28-Search

kaigoos committed:
Search changes

rosasck committed:
merging main

kaigoos commented on simcon_project/users/views/researcher_home.py:
```diff
@@ -1,7 +1,22 @@
+
+
+class ResponseTable(tables.Table):
+    name = tables.columns.Column(accessor="student.get_full_name")
```
We might need to ditch get_full_name and use separate first and last name columns or just email address since when I try to sort the column by alphabetical order by clicking on the column header it seems to not like it. I'll take a look at it a bit and see what I can find first since I think there's no need to separate first and last name if it's possible.

kaigoos commented on simcon_project/users/views/researcher_home.py:
```diff
@@ -1,7 +1,22 @@
+    response = tables.TemplateColumn(
+        '''<a class="btn btn-info btn-sm" href="{% url 'view-response' record.id %}">View</a>''')
+    delete = tables.TemplateColumn(
+        '''<a class="btn btn-danger" href="{% url 'delete-response' record.id %}"  >Delete</a>''')
```
We should either add a verbose name and set it to an empty string, TemplateColumn(verbosename='') and not have a column header or set it to TemplateColumn(orderable=False) to retain the header but have it not be clickable since it results in an error screen right now if the column header is clicked for the template columns.

kaigoos reviewed (dismissed):
The table looks super good and everything works great except for the bugs with the column header sorting stuff I mentioned in the comments.

kaigoos commented on simcon_project/users/views/researcher_home.py:
```diff
@@ -1,7 +1,22 @@
+
+
+class ResponseTable(tables.Table):
+    name = tables.columns.Column(accessor="student.get_full_name")
```
We need an order_by="student.last_name" attribute here so since clicking on the column header to sort causes an error right now. Might be good to change get_full_name in custom-user to change the name format to last name, first name.

devin-w-hill replied:
You could also build the string in researcher_view and pass it in as a dict. Then the accessor is the key you pick

devin-w-hill commented on simcon_project/users/views/researcher_home.py:
```diff
@@ -10,11 +25,28 @@ def is_researcher(user):
+def filter_search(request, responses):
+    if 'searchParam' in request.GET:
+        param = request.GET['searchParam']
+        filter_fields = Q(student__first_name__contains=param) | Q(student__last_name__contains=param) | \
```
I don't know how easy it would be to implement but searching by a student's full name filters out all the responses.

devin-w-hill reviewed (dismissed):
This looks really nice and the search turned out well. Seconded Kai's suggestions and mentioned one minor search thing that I don't think is crucial. Errors on sorting should get fixed before merging at least.

rosasck committed:
Merge branch 'main' of https://github.com/Likhovodov/Simulated-Conversations into Katie/SYMCO28-Search

rosasck committed:
fixing PR comments

rosasck commented:
Fixed PR comments, and added modal thats for some reason being stubborn and turning into a web page.... not sure why... i have work tomorrow so ill debug more tmr!

kaigoos committed:
Modal stuff

rosasck committed:
final modal push

rosasck commented:
added a little "No responses" when the table is empty

Likhovodov reviewed (approved):
Looks good to me

ericharper1 reviewed (changes requested):
This looks awesome! But it seems to not work for full student names. I also have a Convo template called "Conversation 1" and it doesn't show up when I search for it. Maybe it's because of the whitespace?

rosasck commented:
> This looks awesome! But it seems to not work for full student names. I also have a Convo template called "Conversation 1" and it doesn't show up when I search for it. Maybe it's because of the whitespace?

<img width="1267" alt="Screen Shot 2021-02-10 at 10 50 14 PM" src="https://user-images.githubusercontent.com/54516136/107608958-59560780-6bf2-11eb-89a0-a05b0e18a6e0.png">

yeah full name is hard since its two diff params, but u can search for either. and for conversation name it should work! heres mine i named Cintiha Rosas


<img width="1270" alt="Screen Shot 2021-02-10 at 10 54 32 PM" src="https://user-images.githubusercontent.com/54516136/107609154-f3b64b00-6bf2-11eb-8967-983a177b7ac1.png">

ericharper1 reviewed (approved):
Nope, I'm dumb. This is all good!
